src/Network/Email/Envelope.php

```diff
@@ -0,0 +1,65 @@
+<?php
+namespace Cake\Network\Email;
+
+use ArrayAccess;
+
+class Envelope implements ArrayAccess
```

Comment:
Envelope might not be the right name. Envelope has a specific meaning in Email which this isn't. What about TriggeredMail, EmailListener, or DripMail? I don't have a lot of good ideas here.